src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -45,5 +49,284 @@ protected override void InsertItem(int index, IDeltaSetItem item)
 
             base.InsertItem(index, item);
         }
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>.
+        /// </summary>
+        /// <param name="originalCollection">Original collection of the type which needs to be updated.</param>
+        /// /// <returns>DeltaSet response.</returns>
+        public DeltaSet<TStructuralType> Patch(ICollection<TStructuralType> originalCollection)
+        {
+            ODataAPIHandler<TStructuralType> apiHandler = new DefaultODataAPIHandler<TStructuralType>(originalCollection);
+
+            return CopyChangedValues(apiHandler);
+        }
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>.
+        /// </summary>
+        /// <param name="apiHandlerOfT">API Handler for the entity.</param>
+        /// <param name="apiHandlerFactory">API Handler Factory.</param>
+        /// <returns>DeltaSet response.</returns>
+        public DeltaSet<TStructuralType> Patch(ODataAPIHandler<TStructuralType> apiHandlerOfT, ODataAPIHandlerFactory apiHandlerFactory)
+        {         
+            Debug.Assert(apiHandlerOfT != null, "apiHandlerOfT != null");
+
+            return CopyChangedValues(apiHandlerOfT, apiHandlerFactory);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        internal DeltaSet<TStructuralType> CopyChangedValues(IODataAPIHandler apiHandler, ODataAPIHandlerFactory apiHandlerFactory = null)
+        {
+            //Here we are getting the keys and using the keys to find the original object 
+            //to patch from the collection.
+
+            ODataAPIHandler<TStructuralType> apiHandlerOfT = apiHandler as ODataAPIHandler<TStructuralType>;
+
+            Debug.Assert(apiHandlerOfT != null, "apiHandlerOfT != null");
```

Comment:
Out of curiosity, if we expect the `apiHandler` to be an `ODataAPIHandler<TStructuralType>`, why not use that as the parameter type of the method, instead of using the less specific `IODataAPIHandler`?

Reply:
@habbes 
By design. Won't change that for now.